### DISCUSSION TITLE
global file size limit takes precedence over filesizelimit in createBucket api?

### DISCUSSION THREAD

**Discussion by devhandler** [supabase/supabase#15247] [Questions] [answered]
not sure if i understand this correctly, but if global is 50mb and filesizelimit is 100mb, then it makes sense the global limit takes precedence and limit it at 50mb.

but if global is 50mb and filesizelimit is 1mb, does global file size limit still take precedence over the per-bucket file size limit? that is, users can still upload files up to 50mb per file?  that doesn't sound very reasonable. (and you can't adjust global size with free plan). instead, i think it should be the lesser of the two, i.e., min(global limit, per bucket limit)

lmk if i miss sth. thx.

**Comment by GaryAustin1** [comment, marked as answer]
The description (in the js docs at least) seems poorly written.
`specifies the max file size in bytes that can be uploaded to this bucket. The global file size limit takes precedence over this value. The default value is null, which doesn't set a per bucket file size limit.`
It has to be that the filesize in the bucket sets lower limits than the global limit, but not more.
